scipy/sparse/linalg/_norm.py

```diff
@@ -0,0 +1,174 @@
+        compute the vector norms. 
+    
+    Returns
+    -------
+    n : float or matrix
+    
+    Notes
+    -----
+    Some of the ord are not implemented because some associated functions like, 
+    _multi_svd_norm, are not yet available for sparse matrix. 
+    
+    This docstring is modified based on numpy.linalg.norm. 
+    https://github.com/numpy/numpy/blob/master/numpy/linalg/linalg.py 
+    
+    The following norms can be calculated:
+    
+    =====  ============================  
+    ord    norm for sparse matrices             
+    =====  ============================  
+    None   Frobenius norm                
+    'fro'  Frobenius norm                
+    inf    max(sum(abs(x), axis=1))      
+    -inf   min(sum(abs(x), axis=1))      
+    0      abs(x).sum(axis=axis)                           
+    1      max(sum(abs(x), axis=0))      
+    -1     min(sum(abs(x), axis=0))      
+    2      Not implemented  
+    -2     Not implemented      
+    other  Not implemented                               
+    =====  ============================  
+    
+    The Frobenius norm is given by [1]_:
+    
+        :math:`||A||_F = [\\sum_{i,j} abs(a_{i,j})^2]^{1/2}`
+    
+    References
+    ----------
+    .. [1] G. H. Golub and C. F. Van Loan, *Matrix Computations*,
+        Baltimore, MD, Johns Hopkins University Press, 1985, pg. 15
+    
+    Examples
+    --------
+    >>> from scipy.sparse import *
+    >>> import numpy as np
+    >>> from scipy.sparse.linalg import norm
+    >>> a = np.arange(9) - 4
+    >>> a
+    array([-4, -3, -2, -1, 0, 1, 2, 3, 4])
+    >>> b = a.reshape((3, 3))
+    >>> b
+    array([[-4, -3, -2],
+           [-1, 0, 1],
+           [ 2, 3, 4]])
+           
+    >>> b = csr_matrix(b)
+    >>> norm(b)
+    7.745966692414834
+    >>> norm(b, 'fro')
+    7.745966692414834
+    >>> norm(b, np.inf)
+    9
+    >>> norm(b, -np.inf)
+    2
+    >>> norm(b, 1)
+    7
+    >>> norm(b, -1)
+    6
+    
+    Using the `axis` argument to compute vector norms:
+    
+    >>> c = np.array([[ 1, 2, 3],
+    ...               [-1, 1, 4]])
+    >>> c = csr_matrix(c)
+    >>> norm(c, axis=0)
+    matrix[[ 1.41421356, 2.23606798, 5. ]]
+    >>> norm(c, axis=1)
+    matrix[[ 3.74165739, 4.24264069]]
+    >>> norm(c, ord=1, axis=1)
+    matrix[[6]
+           [6]]
+
+"""
+    if not issparse(x):
+        raise TypeError("input is not sparse. use numpy.linalg.norm")
+
+    # Check the default case first and handle it immediately.
+    if ord in [None, 'fro', 'f'] and axis is None:
+        if isComplexType(x.dtype.type):
+            sqnorm = dot(x.real, x.real) + dot(x.imag, x.imag)
+        else:
+            sqnorm = x.power(2).sum()
+        return sqrt(sqnorm)
+
+    # Normalize the `axis` argument to a tuple.
+    nd = x.ndim
+    if axis is None:
+        axis = tuple(range(nd))
+    
+    if np.isscalar(axis):
+        if ord == Inf:
+            return max(abs(x).sum(axis=axis))
```

Comment:
I think this is incorrect.  The sparse norm along a single axis is a vector norm, and the vector infinity norm does not involve a summation.  See https://github.com/numpy/numpy/blob/maintenance/1.9.x/numpy/linalg/linalg.py#L2071.